vm-virtio/src/backend.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright 2019 Intel Corporation. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::cmp;
+use std::io::Write;
+use std::sync::Arc;
+use std::vec::Vec;
+use vmm_sys_util::eventfd::EventFd;
+
+use crate::VirtioInterrupt;
+
+pub struct VirtioBackend {
```

Comment:
@bjzhjing It'd be ideal to be able to share this `VirtioBackend` with all the virtio devices that we have in Cloud-Hypervisor right now, don't you think?
Are you planning to do so? I'm asking because I want to avoid merging this new structure if it only fits the `network` devices.
If it's really too hard to have the same `VirtioBackend` shared across all virtio devices, then maybe a tradeoff would be to call this structure `VirtioNetBackend` instead.

Reply:
/cc @sameo @rbradford

Reply:
@sboeuf Yes, agree! It will be good to have the VirtioBackend shared with all the virtio devices, I will rebase it after #409 merged.

Reply:
Ok sounds good.